### PR TITLE
Eliot.UEExplorer version 1.3.1

### PR DESCRIPTION
**Pull request by EliotVU**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/94562)

### CODE REVIEW

**Comment by EliotVU** [comment]
@microsoft-github-policy-service agree [company=""]

**Comment by EliotVU** [comment]
@microsoft-github-policy-service agree

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: Eliot.UEExplorer
+PackageVersion: 1.3.1
+PackageLocale: en-US
+Publisher: Eliot
+PackageName: UE Explorer
+License: MIT License
```
Comment (mdanish-kh) [inline manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.locale.en-US.yaml:9]: I found this on the Project's GitHub page https://github.com/UE-Explorer/UE-Explorer/blob/UE-Explorer/LICENSE.md which shows the product is licensed under GPL-3.0. Is this a mistake here?

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
The version listed in the Registry/Control Panel/Apps & Features is `1.3.2`, but the installer URL suggests the package is for version `1.3.1`. This seems unintentional, can you please confirm this?

image

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.locale.en-US.yaml (hunk 1 above)
Comment (EliotVU) [reply manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.locale.en-US.yaml]: Oops! :)

**Comment by EliotVU** [comment]
> The version listed in the Registry/Control Panel/Apps & Features is `1.3.2`, but the installer URL suggests the package is for version `1.3.1`. This seems unintentional, can you please confirm this?
> 
> image

Ah, the app is released as 1.3.1 but the setup package was bumped to 1.3.2 because the setup was missing some dependencies. The 1.3.2 setup still installs the 1.3.1 release.

Is that still an issue? Or do I need to make a new setup?

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+PackageIdentifier: Eliot.UEExplorer
+PackageVersion: 1.3.1
+Installers:
+- InstallerLocale: en-US
+  Architecture: x86
+  InstallerType: msi
+  InstallerUrl: https://eliotvu.com/updates/ue-explorer.1.3.1.0.msi
+  InstallerSha256: F1AF3EF606C7E4B0A8694ABA0943840A2C815331FC232FA1904F4F10444A8CA9
+  ProductCode: '{E29ADE45-052E-4CD4-9B13-0D106DA1CB49}'
```
Comment (mdanish-kh) [inline manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml:12]:
```suggestion
  ProductCode: '{E29ADE45-052E-4CD4-9B13-0D106DA1CB49}'
  AppsAndFeaturesEntries:
    - DisplayVersion: 1.3.2
```
WinGet requires the metadata of the package to be consistent with Registry values. If you market the package as `1.3.1`, we can use this in the PackageVersion field which will be the version shown to the user but provide AppsAndFeaturesEntries in the manifest and explicitly provide the `DisplayVersion` (reg value) for WinGet to match upgrades.

Please note that this is only a workaround as this way has the potential to cause an infinite upgrade loop for the user in subsequent releases (or we can have a case where WinGet doesn't upgrade at all to a future `1.3.2` release as this version's DisplayVersion is already `1.3.2`). We would probably need to remove this version with the next release.

My personal recommendation would be to release a new setup fixing this DisplayVersion string written to registry, or if you want this package to be available right now then I'd say to remove this version from the repository when you submit a new version to the repo.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml (hunk 2 above)
Comment (EliotVU) [reply manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml]: Thanks, I have updated the manifest and the installer (at the URL) with the applied changes as per your recommendation.

I also changed the publisher from "Eliot" to "EliotVU" to be consistent in all places.
Just noticed, does this mean I'll also have to rename the /Eliot/ folder hmm..

Comment (mdanish-kh) [reply manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml]: That's not a strict rule. The goal is to make a unique package ID and in WinGet it's generally in the form of `Publisher.PackageName`. It's up to you if you want to do it for consistency.

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: Eliot.UEExplorer
+PackageVersion: 1.3.1
+Installers:
+- InstallerLocale: en-US
+  Architecture: x86
```
Comment (mdanish-kh) [inline manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml:8]:
```suggestion
  Architecture: x64
```
Noticed the architecture in Task Manager. Do we need to change this here?
image

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml (hunk 3 above)
Comment (EliotVU) [reply manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml:8]: Strange, I compiled it targetting x86:
image

File: manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml (hunk 2 above)
Comment (stephengillie) [reply manifests/e/Eliot/UEExplorer/1.3.1/Eliot.UEExplorer.installer.yaml]: The file path, file name, and PackageIdentifier for all 3 files now match Eliot and UEExplorer.

**Comment by stephengillie** [comment]
Hi @EliotUV,

The application errors on first run: 
image
It appears to operate normally after clicking "Continue". Is this expected?

**Comment by EliotVU** [comment]
> Hi @EliotUV,
> 
> The application errors on first run: image It appears to operate normally after clicking "Continue". Is this expected?

Not expected, I did some testing and figured this is caused by missing the prerequisite https://developer.microsoft.com/en-us/microsoft-edge/webview2/ I'll see if I can fix the installer to require. Thanks for reporting that issue btw! I've had someone else ran into this issue as well but I couldn't reproduce till now.

**Comment by EliotVU** [comment]
Upon further investigation, the setup packager is definitely listing the runtime as a prerequisite, however the released .msi setup does not include the steps to install prerequisites, instead we get two files setup.exe and ue-explorer.msi.

Where am I supposed to go from here? Is it possible to package with both the .exe and .msi files?

**Comment by denelon** [comment]
@EliotVU I'm not sure what the steps would be to properly include the dependency in the package. We're getting closer to having the dependency feature working properly. The last lingering issue appears to be:
* microsoft/winget-cli/issues/2939

**Review by stephengillie — APPROVED** [review]

**Comment by stephengillie** [comment]
The application installs and launches normally now.

**Comment by EliotVU** [comment]
Great, thanks for your time!

**Pull request merged**